app/views/merchant/discounts/index.html.erb

```diff
@@ -10,18 +11,21 @@
       <th>Edit</th>
       <th>Delete</th>
     </tr>
-      <% @merchant.discounts.each do |discount| %>
-        <section id="discount-<%=discount.id%>">
-          <tr>
-            <td><%=discount.id %></td>
-            <td><%= link_to discount.name, merchant_discount_path(discount) %></td>
-            <td><%=discount.percent %>%</td>
-            <td><%=discount.id %></td>
-            <td><%= link_to 'Edit', edit_merchant_discount_path(discount) %></td>
-            <td><%= link_to 'Delete', merchant_discount_path(discount), method: :delete %></td>
-          </tr>
-        </section>
-      <%end %>
+        <% @merchant.discounts.each do |discount| %>
+          <% if !discount.id.nil? %>
```

Comment:
tests acted strange. if a discount wasnt saved it still showed up under discounts in RSPEC